ToolBox/Benzinga/BenzingaDataConverter.cs

```diff
@@ -0,0 +1,346 @@
+                        Log.Error($"BenzingaDataConverter.Process(): Failed to map old ticker {symbolTicker}. New ticker is null");
+                        continue;
+                    }
+
+                    instance.Symbols.Add(new BenzingaSymbolData()
+                    {
+                        Exchange = exchange,
+                        Sentiment = sentiment,
+                        Symbol = new Symbol(
+                            SecurityIdentifier.GenerateEquity(symbolTicker, Market.USA, mapSymbol: true, mapFileProvider: _mapFileProvider, mappingResolveDate: instance.PublicationDate),
+                            mappedSymbol
+                        )
+                    });
+                }
+
+                yield return instance;
+            }
+        }
+
+        /// <summary>
+        /// Writes the processed data into their respective files
+        /// </summary>
+        /// <param name="news">Enumerable containing news articles</param>
+        /// <param name="date">Date to convert for</param>
+        private void WriteToFile(IEnumerable<BenzingaNews> news, DateTime date)
+        {
+            var finalDirectory = new DirectoryInfo(Path.Combine(_destinationDirectory.FullName, "content", $"{date:yyyMMdd}"));
+            finalDirectory.Create();
+
+            foreach (var article in news)
+            {
+                var createdReference = false;
+
+                // For each ticker, add a reference to the article we're going to store under the "content" folder
+                foreach (var ticker in article.Symbols)
+                {
+                    // Invalid character in path
+                    if (ticker.Symbol.Value.Contains(":"))
+                    {
+                        Log.Trace($"BenzingaDataConverter.WriteToFile(): Ticker {ticker.Symbol.Value} contains invalid character ':'. Skipping");
+                        continue;
+                    }
+                    // ETFs have a special exchange tag associated with them.
+                    // Sometimes, we have an empty exchange for some tickers. Let's try and map them first before giving up on them.
+                    // Example: SPOT (Spotify) on 2018-10-01 has a missing exchange even though it has already IPO'd
+                    if (ticker.Exchange != "NASDAQ" && ticker.Exchange != "NYSE" && ticker.Exchange != "ETF" && !string.IsNullOrEmpty(ticker.Exchange))
+                    {
+                        Log.Trace($"BenzingaDataConverter.WriteToFile(): Ticker {ticker.Symbol.Value} is not in NYSE, NASDAQ, or is not an ETF. Skipping");
+                        continue;
+                    }
+                    // We can't write these tickers in Windows, so we'll skip them
+                    if (_isWindows && _forbiddenTickers.Contains(ticker.Symbol.Value))
+                    {
+                        Log.Trace($"BenzingaDataConverter.WriteToFile(): Ticker {ticker.Symbol.Value} is invalid in Windows. Skipping");
```

Comment:
Thinking about mapping again -- I guess we don't care _too_ much since we do the data processing in *nix ? Also, thinking about breaking this functionality out into a component, could get it for free

Reply:
With regards to stripping the colon character ":" (since GitHub is glitching out and won't let me reply to your review) - The colon character exclusively appears for international markets, and because we currently don't support other markets, I chose to strip them out. 

And yes, I'd like to split this into another class and refactor, but that's for another PR